packages/utils/src/types/solidity.ts

```diff
@@ -21,9 +20,13 @@ class UINT256 {
   }
 
   static fromProbability(n: number): UINT256 {
-    if (n > 1) throw Error('Probability input cannot be larger than 1')
-    const percent = Math.floor(n * 100)
-    return new UINT256(new BN(constants.MaxUint256.mul(percent).div(100).toString()))
+    if (n > 1) {
+      throw Error('Probability input cannot be larger than 1')
+    }
+
+    const divisor = 10 ** n.toString().replace(/[0]{0,}\./, '').length
```

Comment:
can u replace this with something more readable?

Reply:
added meaningful names and a description what's done

Reply:
Can you add a docstring that describes what this is actually doing - are we simply doing MAXINT * prob ?

Reply:
obsolete now, using inverseProbs